### PR TITLE
method register() was implemented in class RegistrationServiceImpl

### PR DESCRIPTION
**Pull request by mikhail-maiorov**
-tests were added

### CODE REVIEW

**Review by shevalye — APPROVED** [review]

File: src/test/java/core/basesyntax/RegistrationServiceImplTest.java [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+    private static final RegistrationService REGISTRATION = new RegistrationServiceImpl();
+    private static final String EMPTY = "";
+    private static final int CORRECT_AGE = 18;
+    private static final int INCORRECT_AGE = 16;
+    private static final String INCORRECT_LOGIN_WITH_WHITESPACE = "Log in";
+    private static final String CORRECT_PASSWORD = "passWord6";
+    private static final String INCORRECT_PASSWORD_SHORT = "short";
+    private static final String INCORRECT_PASSWORD_WITH_WHITESPACE = "pass Word6";
+    private static final String INCORRECT_PASSWORD_WITHOUT_DIGIT = "passWord";
+    private static final String INCORRECT_PASSWORD_WITHOUT_UPPERCASE = "password6";
```
Comment (dSkarzh) [inline src/test/java/core/basesyntax/RegistrationServiceImplTest.java:24]: just use simple values without creating constants for everything, it's okay in tests
having constant just for minimal required age would be nice, because you could test like MIN_AGE - 1 and MIN_AGE + 1
the rest is redundant

File: src/test/java/core/basesyntax/RegistrationServiceImplTest.java [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+        assertThrows(NullPointerException.class,
+                () -> REGISTRATION.register(FIRST_USER));
+    }
+
+    @Test
+    void register_checkPasswordOnDigit_notOk() {
+        FIRST_USER.setPassword(INCORRECT_PASSWORD_WITHOUT_DIGIT);
+        assertThrows(RuntimeException.class,
+                () -> REGISTRATION.register(FIRST_USER));
+    }
```
Comment (dSkarzh) [inline src/test/java/core/basesyntax/RegistrationServiceImplTest.java:83]:
```suggestion
    @Test
    void register_passwordWithoutDigit_notOk() {
        FIRST_USER.setPassword(INCORRECT_PASSWORD_WITHOUT_DIGIT);
        assertThrows(RuntimeException.class,
                () -> REGISTRATION.register(FIRST_USER));
    }
```

File: src/test/java/core/basesyntax/RegistrationServiceImplTest.java [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+
+    @Test
+    void register_checkPasswordOnWhitespace_notOk() {
+        FIRST_USER.setPassword(INCORRECT_PASSWORD_WITH_WHITESPACE);
+        assertThrows(RuntimeException.class,
+                () -> REGISTRATION.register(FIRST_USER));
+    }
```
Comment (dSkarzh) [inline src/test/java/core/basesyntax/RegistrationServiceImplTest.java:90]:
```suggestion
    @Test
    void register_passwordWithWhitespace_notOk() {
        FIRST_USER.setPassword(INCORRECT_PASSWORD_WITH_WHITESPACE);
        assertThrows(RuntimeException.class,
                () -> REGISTRATION.register(FIRST_USER));
    }
```

File: src/test/java/core/basesyntax/RegistrationServiceImplTest.java [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+
+    @Test
+    void register_checkPasswordOnUppercaseLetter_notOk() {
+        FIRST_USER.setPassword(INCORRECT_PASSWORD_WITHOUT_UPPERCASE);
+        assertThrows(RuntimeException.class,
+                () -> REGISTRATION.register(FIRST_USER));
+    }
```
Comment (dSkarzh) [inline src/test/java/core/basesyntax/RegistrationServiceImplTest.java:97]:
```suggestion
    @Test
    void register_passwordWithoutUppercaseLetter_notOk() {
        FIRST_USER.setPassword(INCORRECT_PASSWORD_WITHOUT_UPPERCASE);
        assertThrows(RuntimeException.class,
                () -> REGISTRATION.register(FIRST_USER));
    }
```

File: src/test/java/core/basesyntax/RegistrationServiceImplTest.java [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+
+    @Test
+    void register_checkPasswordLength_notOk() {
+        FIRST_USER.setPassword(INCORRECT_PASSWORD_SHORT);
+        assertThrows(RuntimeException.class,
+                () -> REGISTRATION.register(FIRST_USER));
+    }
```
Comment (dSkarzh) [inline src/test/java/core/basesyntax/RegistrationServiceImplTest.java:104]:
```suggestion
    @Test
    void register_shortPassword_notOk() {
        FIRST_USER.setPassword(INCORRECT_PASSWORD_SHORT);
        assertThrows(RuntimeException.class,
                () -> REGISTRATION.register(FIRST_USER));
    }
```

File: src/test/java/core/basesyntax/RegistrationServiceImplTest.java [hunk 6]
```diff
@@ -0,0 +1,133 @@
...
+        assertThrows(NullPointerException.class,
+                () -> REGISTRATION.register(FIRST_USER));
+    }
+
+    @Test
+    void register_rangeOfAge_notOk() {
+        FIRST_USER.setAge(INCORRECT_AGE);
+        assertThrows(RuntimeException.class,
+                () -> REGISTRATION.register(FIRST_USER));
+    }
```
Comment (dSkarzh) [inline src/test/java/core/basesyntax/RegistrationServiceImplTest.java:118]:
```suggestion
    @Test
    void register_lessThanMinAge_notOk() {
        User user = new User();
        user.setLogin("validLogin");
        user.setPassword("validPassword123");
        user.setAge(17); // closest to required, but still invalid
        assertThrows(RuntimeException.class,
                () -> registrationService.register(user));
    }
```
this is what good test can look like
also add test for age equal to required, larger than required, negative age

File: src/test/java/core/basesyntax/RegistrationServiceImplTest.java [hunk 7]
```diff
@@ -0,0 +1,133 @@
...
+
+    @Test
+    void register_ageIsNull_notOk() {
+        FIRST_USER.setAge(null);
+        assertThrows(NullPointerException.class,
+                () -> REGISTRATION.register(FIRST_USER));
+    }
+
+    @Test
+    void register_addingExistUser_notOk() {
```
Comment (dSkarzh) [inline src/test/java/core/basesyntax/RegistrationServiceImplTest.java:128]:
```suggestion
    void register_userWithExistingLogin_notOk() {
```

**Review by dSkarzh — CHANGES_REQUESTED** [review]

File: src/test/java/core/basesyntax/RegistrationServiceImplTest.java [hunk 8]
```diff
@@ -0,0 +1,167 @@
...
+        User secondUser = new User();
+        secondUser.setLogin("login");
+        secondUser.setPassword("passWord1");
+        secondUser.setAge(MIN_AGE);
+
+        boolean expect = firstUser.equals(registration.register(firstUser));
+        assertTrue(expect);
+        expect = secondUser.equals(registration.register(secondUser));
+        assertTrue(expect);
+    }
```
Comment (dSkarzh) [inline src/test/java/core/basesyntax/RegistrationServiceImplTest.java:44]: this test in redundant, but you can leave it
```suggestion
    @Test
    void register_additionUsers_ok() {
        User firstUser = new User();
        firstUser.setLogin("Login");
        firstUser.setPassword("passWord1");
        firstUser.setAge(MIN_AGE);

        User secondUser = new User();
        secondUser.setLogin("login");
        secondUser.setPassword("passWord1");
        secondUser.setAge(MIN_AGE);

        boolean expected = firstUser.equals(registration.register(firstUser));
        assertTrue(expect);
        expected = secondUser.equals(registration.register(secondUser));
        assertTrue(expect);
    }
```
change `expect` to `expected` in all tests

File: src/test/java/core/basesyntax/RegistrationServiceImplTest.java [hunk 9]
```diff
@@ -0,0 +1,167 @@
...
+    @Test
+    void register_moreThanMinAge_Ok() {
+        User firstUser = new User();
+        firstUser.setLogin("Login");
+        firstUser.setPassword("passWord1");
+        firstUser.setAge(MIN_AGE + 1);
+
+        boolean expect = firstUser.equals(registration.register(firstUser));
+        assertTrue(expect);
+    }
```
Comment (dSkarzh) [inline src/test/java/core/basesyntax/RegistrationServiceImplTest.java:55]:
```suggestion
    @Test
    void register_moreThanMinAge_Ok() {
        User user = new User();
        user.setLogin("Login");
        user.setPassword("passWord1");
        user.setAge(MIN_AGE + 1);
        boolean expected = user.equals(registration.register(user));
        assertTrue(expected);
    }
```
if there is only one user, don't call it first user, because there is no second one
remove redundant empty line, they take space and your tests are short enough to be read easily
make changes like this in all tests

**Review by dSkarzh — CHANGES_REQUESTED** [review]
Much better!

**Review by dSkarzh — APPROVED** [review]
